frameworks/moveos-stdlib/sources/module_store.move

```diff
@@ -173,6 +203,31 @@ module moveos_std::module_store {
         object::borrow_mut<Allowlist>(allowlist_obj)
     }
 
+    /*********************** package metadata functions *******************/
+    /// Whether a compatibility check should be performed for upgrades. The check only passes if
+    /// a new module has (a) the same public functions (b) for existing resources, no layout change.
+    public fun upgrade_policy_compat(): UpgradePolicy {
+        UpgradePolicy { policy: 1 }
+    }
+
+    /// Whether the modules in the package are immutable and cannot be upgraded.
+    public fun upgrade_policy_immutable(): UpgradePolicy {
+        UpgradePolicy { policy: 2 }
+    }
+
+    /// Whether the upgrade policy can be changed. In general, the policy can be only
+    /// strengthened but not weakened.
+    public fun can_change_upgrade_policy_to(from: UpgradePolicy, to: UpgradePolicy): bool {
+        from.policy <= to.policy
+    }
+
+    public entry fun freeze_package(package: &mut Object<Package>) {
+        let policy: &mut UpgradePolicy = object::borrow_mut_field(package, REVERSED_KEY_UPGRADE_POLICY);
+        policy.policy = upgrade_policy_immutable().policy;
```

Comment:
We can directly freeze the `Object<Package>`.

Reply:
It seems like `UpgradePolicy` is useless right now. If we have `UpgradePolicy` other than `compact` and `immutable`, we can add it in the future.